src/shared/configuration/config.ts

```diff
@@ -109,6 +110,19 @@ export class ApplicationInsightsConfig {
             );
             this.resource = Object.assign(this.resource, options.resource);
             this.samplingRatio = options.samplingRatio || this.samplingRatio;
+
+            // Set console logging level from env var
+            if (process.env[loggingLevel]) { 
+                this.instrumentationOptions = {
+                    ...this.instrumentationOptions,
+                    console: {
+                        ...this.instrumentationOptions.console,
+                        logSendingLevel: logLevelParser(process.env[loggingLevel]),
+                    },
+                }
+            }
+
+            this._mergeConfig();
```

Comment:
This looks like a very risky change, what is going on here with config prioritization? why making this change now?

Reply:
I believe this is effectively a bug today because the `mergeConfig()` method parses the JSON config and environment variables. Env vars need to take precedence over standard in-code configs so that auto instrumentation fixes like setting the `APPLICATIONINSIGHTS_CONFIGURATION_CONTENT` can override any default configs that may already be set in auto-instrumentation scenarios (this is why setting the `console: {enabled: false}` in that env var wasn't working originally).

This also deviates from the way that the distro works, so we end up in a situation where env vars take precedence in the distro (https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/monitor/monitor-opentelemetry/src/shared/config.ts#L111) but not in App Insights. So managed log configurations (winston & bunyan) work as expected while App Insights managed console cannot be correctly overridden by the env var.

Reply:
Sounds good, we need to validate JSON config before releasing agent, this should be a key scenario to support